docs/rfcs/0003-search-ui.md

```diff
@@ -0,0 +1,47 @@
+## Search UI framework
+
+#### Abstract
+Previously, a decision was made to develop a custom elasticsearch UI rather than use searchkit.
+
+From the MIT Open Search Technical Detail doc:
+>We considered using a react component library that constructed queries in the front end, but discarded this approach 
+as not flexible enough for our styling and security needs...and also decided against searchkit due to previous 
+negative experiences using it in MicroMasters.
+
+
+That decision is being revisited due to the increased complexity of the course search UI, which includes facets.
+
+
+#### Architecture Changes
+
+The following enhancements are desired for the course search UI:
+- Default search across all courses if no search text is entered (filtered by facets if selected)
+- Display search result counts next to each facet option
+- Update the list of facets to only show those included in search results
+- Display only the first ~10 topic facets, with a 'More' button to show the entire list
```

Comment:
we also want to display a list of the currently applied filters, like this:

![image](https://user-images.githubusercontent.com/430126/53447610-7b452800-39e3-11e9-804b-8b837a76d1ff.png)